### PR TITLE
Feature/player movement

### PR DESCRIPTION
**Pull request by Guhl23**

### CODE REVIEW

**Comment by PadPalon** [comment]
@Guhl23 I've mostly fixed the bugs, sliding is still kinda fucked on slopes greater than 80 degrees.

@Parolo41 @Hatzman91 I'd appreciate it if you took a look at this instead of me just reviewing as well.

File: Assets/Scripts/Player/CharacterMovementScript.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+using UnityEngine;
+using UnityEngine.AI;
+
+public class CharacterMovementScript : MonoBehaviour {
+    public NavMeshAgent playerNavMeshAgent;
+
+    public Camera playerCamera;
+
+    void Update() {
+        //if the left button of is clicked
```
Comment (Hatzman91) [inline Assets/Scripts/Player/CharacterMovementScript.cs]: left button of what?

File: Assets/Scripts/Player/ThirdPersonMovement.cs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    private Vector3 moveDirection = Vector3.zero;
+
+    //Sliding parameters
+    public float slideSpeed = 10f;
+    private Vector3 hitPointNormal;
+    private bool isSliding = false;
+
+    void Update() {
+        Vector3 direction = CalculateDirection();
```
Comment (Hatzman91) [inline Assets/Scripts/Player/ThirdPersonMovement.cs]: on first glance I have no idea what the direction/moveDirection distinction is for

File: Assets/Scripts/Player/ThirdPersonMovement.cs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    public float jumpHeight = 3f;
+
+    public float turnSmoothTime = 0.1f;
+    private float turnSmoothVelocity;
+
+    private Vector3 moveDirection = Vector3.zero;
+
+    //Sliding parameters
+    public float slideSpeed = 10f;
+    private Vector3 hitPointNormal;
```
Comment (Hatzman91) [inline Assets/Scripts/Player/ThirdPersonMovement.cs]: is there a better name for this?
asked myself if we gave the possum some hitpoints ^^

File: Assets/Scripts/Player/ThirdPersonMovement.cs [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    private bool isSliding = false;
+
+    void Update() {
+        Vector3 direction = CalculateDirection();
+        Vector3 moveDirection = CalculateMoveDirection(direction);
+
+        controller.Move(moveDirection * Time.deltaTime);
```
Comment (Hatzman91) [inline Assets/Scripts/Player/ThirdPersonMovement.cs]: maybe move this into CalculateMoveDirection and rename the method into "MoveInDirection" as this isn't used elsewhere

File: Assets/Scripts/Player/ThirdPersonMovement.cs [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+        if (direction.magnitude >= 0.1f) {
+            float targetAngle = Mathf.Atan2(direction.x, direction.z) * Mathf.Rad2Deg;
+            float angle = Mathf.SmoothDampAngle(transform.eulerAngles.y, targetAngle, ref turnSmoothVelocity, turnSmoothTime);
+            transform.rotation = Quaternion.Euler(0f, angle, 0f);
+        }
+    }
+
+    private Vector3 CalculateDirection() {
+        if (isSliding) {
```
Comment (Hatzman91) [inline Assets/Scripts/Player/ThirdPersonMovement.cs:29]: isSliding is checked twice in consecutive methods, might be an issue if either code gets bigger --> confusion

File: Assets/Scripts/Player/ThirdPersonMovement.cs (hunk 2 above)
Comment (PadPalon) [reply Assets/Scripts/Player/ThirdPersonMovement.cs]: The first is the normalized direction the character wants to move in (either through input or forced sliding), the second is the actual vector along which the movement for the next frame happens (with clamped speed for the xz magnitued as well as changes in elevation through jumping or gravity)

Find clearer naming

File: Assets/Scripts/Player/ThirdPersonMovement.cs (hunk 4 above)
Comment (PadPalon) [reply Assets/Scripts/Player/ThirdPersonMovement.cs]: The intent was to keep these methods as pure as possible, but there is actually a mistake concering purity here. CalculateMoveDirection should not operate on the moveDirection member variable, but get it passed as an argument as to limit the side effects.

**Review by Parolo41 — COMMENTED** [review]
Didn't see anything additional. Ditto on the naming clarification.

**Review by Hatzman91 — APPROVED** [review]

**Pull request merged**